client/jsonrpc/client.go

```diff
@@ -0,0 +1,117 @@
+package jsonrpc
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"sync"
+
+	"github.com/duneanalytics/blockchain-ingester/lib/hexutils"
+	"github.com/duneanalytics/blockchain-ingester/models"
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+type BlockchainClient interface {
+	LatestBlockNumber() (int64, error)
+	BlockByNumber(ctx context.Context, blockNumber int64) (models.RPCBlock, error)
+
+	// SendBlocks sends blocks from startBlockNumber to endBlockNumber to outChan, inclusive.
+	// If endBlockNumber is -1, it sends blocks from startBlockNumber to the tip of the chain
+	// it will run continueously until the context is cancelled
+	SendBlocks(ctx context.Context, outChan chan models.RPCBlock, startBlockNumber, endBlockNumber int64) error
+
+	Close() error
+}
+
+const (
+	MaxRetries = 10
+)
+
+type rpcClient struct {
+	client  *retryablehttp.Client
+	cfg     Config
+	log     *slog.Logger
+	bufPool *sync.Pool
+}
+
+func NewRPCClient(cfg Config, log *slog.Logger) *rpcClient { // revive:disable-line:unexported-return
+	client := retryablehttp.NewClient()
+	client.RetryMax = MaxRetries
+	client.Logger = log
+	client.CheckRetry = retryablehttp.DefaultRetryPolicy
+	client.Backoff = retryablehttp.LinearJitterBackoff
+	return &rpcClient{
+		client: client,
+		cfg:    cfg,
+		log:    log,
+		bufPool: &sync.Pool{
+			New: func() interface{} {
+				return new(bytes.Buffer)
+			},
+		},
+	}
+}
+
+func (c *rpcClient) LatestBlockNumber() (int64, error) {
+	buf := c.bufPool.Get().(*bytes.Buffer)
+	defer c.bufPool.Put(buf)
+	buf.Reset()
+
+	err := c.getResponseBody(context.Background(), "eth_blockNumber", []any{}, buf)
+	if err != nil {
+		c.log.Error("Failed to get response for jsonRPC",
+			"method", "eth_blockNumber",
+			"error", err,
+		)
+		return 0, err
+	}
+	resp := struct {
+		Result string `json:"result"`
+	}{}
+	if err := json.NewDecoder(buf).Decode(&resp); err != nil {
+		c.log.Error("Failed to decode response for jsonRPC", "error", err)
+		return 0, err
+	}
+	return hexutils.IntFromHex(resp.Result)
+}
+
+// getResponseBody sends a request to the OpStack server and returns the response body
```

Comment:
```suggestion
// getResponseBody sends a request to the server and returns the response body
```
no mention of opstack elsewhere I think?